cdap-data-fabric/src/main/java/co/cask/cdap/data2/nosql/NoSqlTransactionRunner.java

```diff
@@ -45,16 +45,17 @@
   @Inject
   public NoSqlTransactionRunner(NoSqlStructuredTableAdmin tableAdmin, TransactionSystemClient txClient) {
     this.tableAdmin = tableAdmin;
-    this.transactional = Transactions.createTransactionalWithRetry(createTransactional(txClient),
-                                                                   RetryStrategies.retryOnConflict(20, 100));
+    this.transactional = Transactions.createTransactionalWithRetry(
+      createTransactional(txClient, tableAdmin::getEntityTable),
+      RetryStrategies.retryOnConflict(20, 100));
   }
 
-  private Transactional createTransactional(TransactionSystemClient txClient) {
+  static Transactional createTransactional(TransactionSystemClient txClient, TableDatasetAccesor datasetAccesor) {
```

Comment:
Please don't mix static method to non util type class (especially this class is Guice inject). Make this a util method in util class in the same package.

Reply:
Done